src/workerd/api/gpu/gpu-buffer.c++

```diff
@@ -46,7 +51,23 @@ GPUBuffer::getMappedRange(jsg::Optional<GPUSize64> offset,
 
   JSG_REQUIRE(ptr, TypeError, "could not obtain mapped range");
 
-  auto arrayBuffer = kj::arrayPtr((byte *)ptr, s);
+  struct Context {
+    jsg::Ref<GPUBuffer> buffer;
+  };
+  auto ref = JSG_THIS;
+  auto ctx = new Context{ref.addRef()};
+  std::shared_ptr<v8::BackingStore> backing = v8::ArrayBuffer::NewBackingStore(
+      ptr, s,
+      [](void *data, size_t length, void *deleter_data) {
+        // we have a persistent reference to GPUBuffer so that it lives at least
+        // as long as this arraybuffer
+        auto c = std::unique_ptr<Context>(static_cast<Context *>(deleter_data));
```

Comment:
@kentonv ... I think it's helpful to have your eyes on this also. Do you see any risk in having the `v8::BackingStore` holding the reference to `jsg::Ref` like this? Unfortunately with this particular case we would definitely need to hold something to ensure that the `v8::BackingStore` remains safe to access.

Reply:
FTR, Chromium does something similar with the empty deleter: https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:third_party/blink/renderer/modules/webgpu/gpu_buffer.cc;l=415;drc=0340dd86c6c1f01ccd1a0b983633cfef5fcf6697

Reply:
Maybe @harrishancock can weigh in as well. :)

Reply:
The chromium code there does show a more sophisticated approach that is likely worthwhile to emulate. Specifically, the `ArrayBuffer` is created with a specific "detach key" that prevents the `v8::BackingStore` from being detached without the matching key, largely preventing the kind of issue I'm concerned about here. You can also see that they are keeping track of the `ArrayBuffer` instances they create and are proactively detaching them when the `GPUBuffer` is destroyed. The combination of those two may make it "safe enough" for most cases but I still prefer to keep the owning `GPUBuffer` reference around as part of the `v8::BackingStore` given that our `kj::Array<kj::byte>` type-handler will point at the `v8::BackingStore`'s memory buffer without detaching it.

Reply:
Done, PTAL.